cmake/modules/hunter_add_repository.cmake

```diff
@@ -0,0 +1,81 @@
+include(CMakeParseArguments) # cmake_parse_arguments
+include(hunter_internal_error)
+include(hunter_fatal_error)
+include(hunter_test_string_not_empty)
+include(hunter_status_debug)
+include(hunter_status_print)
+include(hunter_install_repository)
+
+function(hunter_add_repository)
+  if(PROJECT_NAME OR NOT HUNTER_GATE_ROOT)
```

Comment:
> this is also the behavior for HunterGate?

Only one `HunterGate` parameters will be used. Next called `HunterGate` will do nothing, effectively using settings from the very first.

> Does that sound right?

Should be used oncd - yes. Can be called once - no. You should detect that settings from `hunter_add_repository` already used by Hunter and next calls to `hunter_add_repository` should be ignored.